### DISCUSSION TITLE
The correct validation schema for a numeric field based on the number of characters.

### DISCUSSION THREAD

**Discussion by yudinmaxim** [open-circle/valibot#394] [Q&A]
Hello everyone. I've encountered a problem with typing a form validation schema, where there's a numeric field that requires checking the number of entered characters. I found the following solution: 
```javascript
const deviceSchema = transform(
  object({
    id: string(),
    index: number(),
    imsi: coerce(string([
      minLength(1, 'IMSI need value'),
      minLength(12, 'IMSI need min length 12'),
      maxLength(15, 'IMSI need max length 15'),
      regex(/^[0-9]+$/, 'IMSI must be a number'),
    ]),
    (input) => {
      console.log('Coerce: ', input)
      return input?.toString()
    })
  }),
  (input) => {
    return {
      ...input,
      imsi: Number(input.imsi)
    }
  }
)
```

And it even works) But it looks odd.

I'm using a validator together with vee-validate and I'm getting a type error on the line:
```javascript
const onFormSubmit = handleSubmit((values) => {
  console.log('onFormSubmit', { values })

  emit('update:model-value', values) // linter type error for values
  emit('save')
})
```

handleSubmit is a validation method from vee-validate.
The fields of the schema are based on the iDevice interface.
From the schema, I derive the type:
```javascript
type tDevice = schemaType<typeof deviceSchema>;
```
I expect that tDevice overlaps with iDevice enough for a value of type tDevice to be valid for the type iDevice.

What is the most accurate way to validate the number of characters in a numeric field? I think my solution is incorrect (even though it works) and there's a proper way to do it.

Thank you for any assistance.

**Comment by yudinmaxim** [comment]
I have found a solution. I changed the regular minLength and maxLength methods and they work the way I need them to

**Reply by fabian-hiller** [reply]
We could add an `imsi` validation function to the library. Also, whenever the existing validation functions are not sufficient for your use case, you can implement custom logic with `custom`.

**Reply by yudinmaxim** [reply]
Than you for you answer.
The main problem with validation for imsi was that it was necessary to count the number of characters. At the moment, I have solved this by redesigning the function for determining the number of characters in the field value.
If you make a separate validation function for `imsi`, then it is not very clear to me how to make a description of various errors:
- the length is less than necessary;
- the length is more than necessary;
- the characters in the field value are invalid.
Can a single validator accept a series of error variants?

**Reply by fabian-hiller** [reply]
Yes, in this chase, we will only output a single error message, but it is technically possible to output multiple messages.
